specs/experimental/contracts/safe/liveness-guard.md

```diff
@@ -0,0 +1,189 @@
+# LivenessGuard
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [Definitions](#definitions)
+  - [Liveness Timestamp](#liveness-timestamp)
+  - [Owners Before Set](#owners-before-set)
+- [Assumptions](#assumptions)
+  - [a01-001: Safe Contract Integrity](#a01-001-safe-contract-integrity)
+    - [Mitigations](#mitigations)
+- [Invariants](#invariants)
+  - [i01-001: Signature Attribution Accuracy](#i01-001-signature-attribution-accuracy)
+    - [Impact](#impact)
+  - [i02-002: Non-Reversion Guarantee](#i02-002-non-reversion-guarantee)
+    - [Impact](#impact-1)
+  - [i03-003: Owner Set Synchronization](#i03-003-owner-set-synchronization)
+    - [Impact](#impact-2)
+- [Function Specification](#function-specification)
+  - [constructor](#constructor)
+  - [safe](#safe)
+  - [checkTransaction](#checktransaction)
+  - [checkAfterExecution](#checkafterexecution)
+  - [showLiveness](#showliveness)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+The LivenessGuard tracks the liveness of Safe multisig owners by recording timestamps when owners sign transactions or
+explicitly demonstrate activity. It serves as a monitoring mechanism to identify inactive owners who may have lost
+access to their keys.
+
+## Definitions
+
+### Liveness Timestamp
+
+The most recent block timestamp at which an owner either signed a Safe transaction or called the `showLiveness`
+function. This timestamp is publicly queryable via the `lastLive` mapping.
+
+### Owners Before Set
```

Comment:
devinai Why are you defining implementation details? Implementation details should not be part of definitions